db/migrations/89_mark_attempts_as_changed_on_groups_relations_adding.sql

```diff
@@ -0,0 +1,108 @@
+-- +migrate Up
+-- +migrate StatementBegin
+CREATE TRIGGER `after_insert_groups_groups` AFTER INSERT ON `groups_groups` FOR EACH ROW BEGIN
+    IF NEW.`expires_at` > NOW() THEN
+        UPDATE `attempts`
+            JOIN `groups_ancestors_active` ON `groups_ancestors_active`.`child_group_id` = `attempts`.`group_id` AND
+                                              `groups_ancestors_active`.`ancestor_group_id` = NEW.`child_group_id`
+        SET `result_propagation_state` = 'changed'
+        WHERE EXISTS(
+            SELECT 1 FROM `permissions_generated`
+                JOIN `groups_ancestors_active` AS `grand_ancestors`
+                    ON `grand_ancestors`.`child_group_id` = NEW.`parent_group_id` AND
+                       `grand_ancestors`.`ancestor_group_id` = `permissions_generated`.`group_id`
+                JOIN `items_ancestors`
+                    ON `items_ancestors`.`ancestor_item_id` = `permissions_generated`.`item_id`
+            WHERE `items_ancestors`.`child_item_id` = `attempts`.`item_id` AND
+                  `permissions_generated`.`can_view_generated` != 'none'
+        ) AND NOT EXISTS(
+            SELECT 1 FROM `permissions_generated`
+                JOIN `groups_ancestors_active` AS `child_ancestors`
+                    ON `child_ancestors`.`child_group_id` = NEW.`child_group_id` AND
+                       `child_ancestors`.`ancestor_group_id` = `permissions_generated`.`group_id`
+                JOIN `items_ancestors`
+                    ON `items_ancestors`.`ancestor_item_id` = `permissions_generated`.`item_id`
+            WHERE `items_ancestors`.`child_item_id` = `attempts`.`item_id` AND
+                  `permissions_generated`.`can_view_generated` != 'none'
+        );
+    END IF;
+END
+-- +migrate StatementEnd
+
+DROP TRIGGER `before_update_groups_groups`;
+-- +migrate StatementBegin
+CREATE TRIGGER `before_update_groups_groups` BEFORE UPDATE ON `groups_groups` FOR EACH ROW BEGIN
+    IF (OLD.child_group_id != NEW.child_group_id OR OLD.parent_group_id != NEW.parent_group_id OR OLD.expires_at != NEW.expires_at) THEN
```

Comment:
Again the same question, do we really want to support changing parent/child in groups_groups?

Reply:
fixed